src/radix_connect/p2p_links/link_connection_qr_data.rs

```diff
@@ -0,0 +1,173 @@
+    fn sample() -> Self {
+        Self::new(
+            RadixConnectPurpose::sample(),
+            RadixConnectPassword::sample(),
+            Ed25519PublicKey::sample(),
+            Exactly64Bytes::sample(),
+        )
+    }
+
+    fn sample_other() -> Self {
+        Self::new(
+            RadixConnectPurpose::sample_other(),
+            RadixConnectPassword::sample_other(),
+            Ed25519PublicKey::sample_other(),
+            Exactly64Bytes::sample_other(),
+        )
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::prelude::*;
+
+    #[test]
+    fn equality() {
+        assert_eq!(
+            LinkConnectionQRData::sample(),
+            LinkConnectionQRData::sample()
+        );
+        assert_eq!(
+            LinkConnectionQRData::sample_other(),
+            LinkConnectionQRData::sample_other()
+        );
+    }
+
+    #[test]
+    fn inequality() {
+        assert_ne!(
+            LinkConnectionQRData::sample(),
+            LinkConnectionQRData::sample_other()
+        );
+    }
+
+    #[test]
+    fn debug() {
+        assert_eq!(
+            format!("{:?}", LinkConnectionQRData::sample()),
+            "LinkConnectionQRData { purpose: 'general', password: 'deaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddead', public_key: 'ec172b93ad5e563bf4932c70e1245034c35467ef2efd4d64ebf819683467e2bf', signature: 'deaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddead' }"
+        );
+    }
+
+    #[test]
+    fn display() {
+        assert_eq!(
+            format!("{}", LinkConnectionQRData::sample()),
+            "LinkConnectionQRData with purpose: 'general'"
+        );
+    }
+
+    #[test]
+    fn json_roundtrip() {
+        let sut = LinkConnectionQRData::sample();
+        assert_eq_after_json_roundtrip(
+            &sut,
+            r#"
+            {
+                "purpose": "general",
+                "password": "deaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddead",
+                "publicKey": "ec172b93ad5e563bf4932c70e1245034c35467ef2efd4d64ebf819683467e2bf",
+                "signature": "deaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddead"
+            }
+            "#,
+        );
+    }
+
+    #[test]
+    fn purpose() {
+        assert_eq!(
+            LinkConnectionQRData::sample().purpose,
+            RadixConnectPurpose::General
+        );
+    }
+
+    #[test]
+    fn password() {
+        assert_eq!(
+            LinkConnectionQRData::sample().password.hash().to_string(),
+            "9059d2ac799749e2f9f18541015197051ff9f803741d566744fe34ea004a5908"
+        );
+    }
+}
+
+#[cfg(test)]
+mod uniffi_tests {
+    use crate::prelude::*;
+
+    #[test]
+    fn sample_values() {
+        assert_eq!(
+            new_link_connection_qr_data_sample(),
+            LinkConnectionQRData::sample()
```

Comment:
Prefer `SUT` and maybe you can use hash, like `sargon_build_information_uniffi_fn`:

```rust
    #[test]
    fn hash_of_samples() {
        assert_eq!(
            HashSet::<SUT>::from_iter([
                new_sargon_build_information_sample(),
                new_sargon_build_information_sample_other(),
                // duplicates should get removed
                new_sargon_build_information_sample(),
                new_sargon_build_information_sample_other(),
            ])
            .len(),
            2
        );
    }
```